src/components/RegistrationDetailsPage.js

```diff
@@ -154,8 +155,42 @@ const RegistrationAnswers = ({ questions }) => {
   )
 }
 
+const GroupDetails = ({ groupDetails }) => {
+  if (!groupDetails) {
+    return (
+      <div>
+        <h2>This user is currently not part of any group.</h2>
+      </div>
+    )
+  } else {
+    return (
+      <div>
+        <h2>{groupDetails.groupName}</h2>
+        {groupDetails.students.map((member, index) => {
+          return (
+            <p>
+              {index + 1}. {member.first_names} {member.last_name}
+            </p>
+          )
+        })}
+        <h3>Ohjaaja</h3>
+        <p>{groupDetails.instructor}</p>
+      </div>
+    )
+  }
+}
+
 class RegistrationDetailsPage extends React.Component {
+  state = { groupDetails: null }
+
+  async componentDidMount() {
+    const myGroup = await groupManagementService.getByStudent()
+    this.setState({
```

Comment:
Why is this in local state and not in redux state? Redux state is still passed in via mapStateToProps but it's never used?

Reply:
instructor attribute could be fetched on login and stored to redux state, so it does not need to be refetched from backend on every page.